strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/jwt/JWTAuthenticator.java

```diff
@@ -0,0 +1,43 @@
+package org.carlspring.strongbox.authentication.api.impl.jwt;
+
+import org.carlspring.strongbox.authentication.api.AuthenticationSupplier;
+import org.carlspring.strongbox.authentication.api.Authenticator;
+import org.carlspring.strongbox.authentication.api.impl.userdetails.StrongboxUserDetailService;
+import org.carlspring.strongbox.users.security.SecurityTokenProvider;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class JWTAuthenticator
+        implements Authenticator
+{
+
+    @Inject
+    @StrongboxUserDetailService.StrongboxUserDetailServiceQualifier
+    private UserDetailsService userDetailsService;
+
+    @Inject
+    private SecurityTokenProvider securityTokenProvider;
+
+    @Nonnull
+    @Override
+    public AuthenticationProvider getAuthenticationProvider()
+    {
+        return new JWTAuthenticationProvider(userDetailsService, securityTokenProvider);
```

Comment:
I agree, it can be, but I can't see a big profit here. 

Authentication Providers, authentication suppliers, user management components - it all will be scanned by the child generic application context, as we agreed. The only entry point to those things will be authentication registry component registered in the root application context. So in my opinion, keeping JWTAuthenticationProvider as a plain object gives me simpler way to unit test this provider.